src/super_gradients/convert_from_recipe.py

```diff
@@ -0,0 +1,206 @@
+
+
+def import_black_or_fail_with_instructions():
+    try:
+        import black
+
+        return black
+    except ImportError:
+        raise ImportError(
+            "Black is not installed. Please install it with `pip install black` and try again. "
+            "If you are using a virtual environment, make sure it is activated."
+        )
+
+
+def import_black_or_install():
+    try:
+        import black
+
+        return black
+    except ImportError:
+        import pip
+
+        pip.main(["install", "black==22.10.0"])
+
+        return import_black_or_fail_with_instructions()
+
+
+black = import_black_or_install()
+
+
+def recursively_walk_and_extract_hydra_targets(cfg, objects=None, prefix=None) -> Tuple[DictConfig, Dict[str, Mapping]]:
+    if objects is None:
+        objects = collections.OrderedDict()
+    if prefix is None:
+        prefix = ""
+
+    if isinstance(cfg, DictConfig):
+        for key, value in cfg.items():
+            value, objects = recursively_walk_and_extract_hydra_targets(value, objects, prefix=f"{prefix}_{key}")
+            cfg[key] = value
+
+        if "_target_" in cfg:
+            target_class = cfg["_target_"]
+            target_params = dict([(k, v) for k, v in cfg.items() if k != "_target_"])
+            object_name = f"{prefix}".replace(".", "_").lower()
+            objects[object_name] = (target_class, target_params)
+            cfg = object_name
+
+    elif isinstance(cfg, ListConfig):
+        for index, item in enumerate(cfg):
+            item, objects = recursively_walk_and_extract_hydra_targets(item, objects, prefix=f"{prefix}_{index}")
+            cfg[index] = item
+    else:
+        print(f"Skipping {cfg}")
+        pass
+    return cfg, objects
+
+
+def convert_from_recipe(cfg: DictConfig, output_script_path: str):
+    cfg = Trainer._trigger_cfg_modifying_callbacks(cfg)
+    OmegaConf.resolve(cfg)
+
+    device = get_param(cfg, "device")
+    multi_gpu = get_param(cfg, "multi_gpu")
+
+    if multi_gpu is False:
+        multi_gpu = MultiGPUMode.OFF
+    num_gpus = get_param(cfg, "num_gpus")
+    print(device, multi_gpu, num_gpus)
+
+    train_dataloader = get_param(cfg, "train_dataloader")
+    train_dataset_params = OmegaConf.to_container(cfg.dataset_params.train_dataset_params, resolve=True)
+    train_dataloader_params = OmegaConf.to_container(cfg.dataset_params.train_dataloader_params, resolve=True)
+
+    val_dataloader = get_param(cfg, "val_dataloader")
+    val_dataset_params = OmegaConf.to_container(cfg.dataset_params.val_dataset_params, resolve=True)
+    val_dataloader_params = OmegaConf.to_container(cfg.dataset_params.val_dataloader_params, resolve=True)
+
+    num_classes = cfg.arch_params.num_classes
+    arch_params = OmegaConf.to_container(cfg.arch_params, resolve=True)
+
+    training_hyperparams, hydra_instantiated_objects = recursively_walk_and_extract_hydra_targets(cfg.training_hyperparams)
+    print(hydra_instantiated_objects)
```

Comment:
I guess this was left here unintentionally.

Reply:
Indeed. Removed